src/plugins/accounting.cpp

```diff
@@ -10,3 +10,28 @@
 
 #include "accounting.hpp"
 
+Association* get_association (int userid,
+                              const char *bank,
```

Comment:
Why not take a `std::string` bank and return an `Association&` rather than a pointer? You'd need to throw an exception instead of returning `nullptr` though I think.

Reply:
The reason for using `char *` instead of `std::string` is because the type of `bank` when it is unpacked with `flux_plugin_arg_unpack ()` is of type `char *`, and (to my limited C++ knowledge) there isn't an easy way of converting it to `std::string`.

The reason I chose to `return` a pointer to an `Association` object is because the existence of an `Association` object is not guaranteed (i.e there could be a user or user/bank combo that does not exist in the flux-accounting DB), and the plugin needs to handle this case, which I don't _believe_ is easily handled if I `return` a reference (I could be wrong though).

Reply:
Couldn't you do something like pass `std::string{bank}` when you invoke the function, and then take a const reference to it in the function invocation?

And yeah in the case where the object doesn't exist, I don't think you could return a null reference, you'd need to throw something like `std::out_of_range`. 

Anyway, it's fine as it is, it's just my understanding that using raw pointers in C++ is not recommended except in special cases. But I'm not confident in my C++ style so I won't press the point, I think it's good to be merged!

Reply:
Thanks! I can look into this further in the near future. My proposal here was just to keep things as similar to what they were beforehand as possible so that it required very little changes within the plugin code itself 😆 . Will set MWP here